src/Core/Scheduler/Providers/GpxUtils.js

```diff
@@ -16,78 +20,173 @@ function _gGpxToWTrackPointsArray(gpxXML) {
     return gpxXML.getElementsByTagName('trkpt');
 }
 
-function _gpxPtToCartesian(pt) {
+function _gpxPtToCartesian(pt, crs) {
     var longitude = Number(pt.attributes.lon.nodeValue);
     var latitude = Number(pt.attributes.lat.nodeValue);
-    var elevation = Number(pt.getElementsByTagName('ele')[0].childNodes[0].nodeValue);
+    // TODO: get elevation with terrain
+    const elem = pt.getElementsByTagName('ele')[0];
+    const elevation = elem ? Number(elem.childNodes[0].nodeValue) : 0;
 
-    return new Coordinates('EPSG:4326', longitude, latitude, elevation).as('EPSG:4978').xyz();
+    return new Coordinates('EPSG:4326', longitude, latitude, elevation).as(crs).xyz();
 }
 
-function _gpxToWayPointsMesh(gpxXML) {
+const geometryPoint = new THREE.BoxGeometry(1, 1, 80);
+const materialPoint = new THREE.MeshBasicMaterial({ color: 0xffffff });
+const point = new THREE.Vector3();
+
+function getDistance(object, camera) {
+    point.setFromMatrixPosition(object.matrixWorld).add(object.geometry.boundingSphere.center);
+    return camera.position.distanceTo(point);
+}
+
+function updatePointScale(renderer, scene, camera) {
+    const distance = getDistance(this, camera);
+    const scale = Math.max(2, Math.min(100, distance / renderer.getSize().height));
+    this.scale.set(scale, scale, scale);
+    this.updateMatrixWorld();
+}
+
+function _gpxToWayPointsMesh(gpxXML, crs) {
     var wayPts = _gpxToWayPointsArray(gpxXML);
 
     if (wayPts.length) {
-        const positions = new Float32Array(wayPts.length * 3);
-        for (var i = 0; i < wayPts.length; i++) {
-            const pos = _gpxPtToCartesian(wayPts[i]);
-            pos.toArray(positions, 3 * i);
+        const points = new THREE.Group();
+
+        if (gpxXML.center === undefined) {
+            gpxXML.center = _gpxPtToCartesian(wayPts[0], crs);
+        }
+
+        const lookAt = gpxXML.center.clone().negate();
+
+        for (const wayPt of wayPts) {
+            const position = _gpxPtToCartesian(wayPt, crs).sub(gpxXML.center);
+            // use Pin to make it more visible
+            const mesh = new THREE.Mesh(geometryPoint, materialPoint);
+            mesh.position.copy(position);
+            mesh.lookAt(lookAt);
+
+            // Scale pin in function of distance
+            mesh.onBeforeRender = updatePointScale;
+
+            points.add(mesh);
         }
-        const points = new THREE.Points();
-        points.geometry.addAttribute('position', new THREE.BufferAttribute(positions, 3));
-        points.material.color.setRGB(0, 1, 0);
         return points;
     } else {
         return null;
     }
 }
 
-function _gpxToWTrackPointsMesh(gpxXML) {
+function updatePath(renderer, scene, camera) {
+    const distance = getDistance(this, camera);
+    this.material.depthTest = distance < this.geometry.boundingSphere.radius * 2;
```

Comment:
Why do you need to disable depth test?
Is there a valid reason or is this a workaround for a bug in iTowns?